_blog/2023-10-11-graphql-in-rust.mdx

```diff
@@ -0,0 +1,492 @@
+---
+title: 'Using GraphQL in Rust'
+description: This article details how to build a GraphQL server in Rust. We will explore using queries, mutations as well as subscriptions to make a fully working endpoint.
+author: josh
+tags: [rust, graphql, sql, guide]
+thumb: graphql-in-rust-thumb.png 
+cover: graphql-in-rust-thumb.png 
+date: '2023-10-04T14:30:00'
+---
+# Building a GraphQL Server in Rust
+When it comes to writing an API, sometimes you might have several data sources and want to coalesce them into one easy-to-query API on the frontend. This is where GraphQL comes in: an query language made for APIs and declarative data fetching (you only query what you want). Here are some advantages that GraphQL can bring to your Rust web application:
+- Test your queries out in real-time via the GraphQL playground
+- Makes it much easier for your frontend to query your backend
+- You can use any data source
+
+In this example, we will use GraphQL through the `async-graphql` Rust crate as an Axum endpoint with an SQL data source and we'll be creating an API that can create, update, and delete a table of records about dogs, as well as subscribing to any updates.
+
+Stuck or want to know what the final code looks like? [You can find the repository here.](https://github.com/joshua-mo-143/shuttle-axum-gql-ex)
+
+# Getting Started
+
+You'll want to initiate a new Shuttle project (requires `cargo-shuttle`):
+```bash
+cargo shuttle init
+```
+
+For this article we'll be using the project name "graphql-example". When the CLI asks you what framework you want, pick Axum.
```

Comment:
Maybe remind about unique name? Well, the CLI does deny duplicate names now so whatever